site/_includes/macros/cards/blog-card.njk

```diff
@@ -3,12 +3,8 @@
 {% macro blogCard(post) %}
 {% set pd = post.data %}
 <div class="blog-card rounded-lg pad-300 md:pad-400 width-full">
-  {% if pd.thumbnail %}
-  {% set thumbnail = pd.thumbnail %}
-  {% else %}
-  {% set thumbnail = pd.hero %}
-  {% endif %}
-  {% set alt = pd.alt %}
+  {% set thumbnail = pd.thumbnail or pd.hero or site.defaultPostImg %}
+  {% set alt = pd.alt or 'i18n.common.blog_default_alt' | i18n(locale) %}
```

Comment:
You need to make a condition on whether the thumbnail is defined, not alt. Otherwhise you set a default alt on a non-default thumbnail:

![image](https://user-images.githubusercontent.com/1914261/223147587-87f0f61e-189a-4804-926d-209aaf2c8bae.png)